### DISCUSSION TITLE
Discourse Reactivated

### DISCUSSION THREAD

**Discussion by abstractj** [keycloak/keycloak#21112] [Announcements]
Our community has grown and evolved significantly over time. This growth has 
been exciting to be a part of, but it has also presented new challenges. One of 
these challenges has been managing the multiple communication channels we 
have in place to engage with our community.

Today, we have various channels available: Google Groups, GitHub Discussions, 
Slack, and Discourse. Each of these platforms has its strengths in fostering the 
open dialogue and collaboration. However, we have been unable to provide the 
level of attention and interaction we believe our community deserves.

With this in mind, we have decided that as of June 26, 2023, we will sunset 
Discourse as one of our communication channels. We understand the value 
that Discourse has brought to our community, but we believe that focusing our 
efforts on fewer communication channels will allow us to foster more 
meaningful conversations.

What does this mean for Discourse users?

From June 26, 2023, users will no longer be able to create new posts on 
Discourse, but they can still access all the content and discussions that have 
been shared there. All the existing threads will be preserved as an archive, 
ensuring that the knowledge accumulated over the years is not lost.

For those who have been actively using Discourse, we understand that this 
transition might be challenging. We encourage users to join our conversations 
on [Google Groups, GitHub Discussions, and Slack](https://www.keycloak.org/community).

**Comment by xgp** [comment]
Thank you for attempting to meet the spirit of [this request](https://github.com/keycloak/keycloak/discussions/20861). It’s rare that a meta-issue about the community is actually heard here, so I appreciate at least that. 

As usual, I’m disappointed with the lack of discussion, data, transparency and community engagement in this decision. A number of community members placed a huge amount of effort and time in crafting high-fidelity, respectful answers to support questions on Discourse that weren’t being answered elsewhere. This had the net effect of **reducing** the support burden on the maintainers / RedHat employees (the exclusive “we” you refer to). Furthermore, Discourse topics seem to be vastly better optimized for search engine results, once again **reducing** repeat questions and thus engagement burden. 

If the goal is to free your attention, this may achieve it, but it fails to help the community as a whole have more meaningful interactions and build permanent, discoverable content. I know that I and others were really expecting a more inclusive “we” following Keycloak’s acceptance into CNCF.

**Reply by jonkoops** [reply]
As Bruno mentioned, we will be making the Discourse forum read-only. This means that all of the content that was there, including those answers, will remain accessible.

**Comment by xgp** [comment]
Thanks @jonkoops. I understand that point. But what about clarifications and future discussions? For example, there were a bunch of older topics there that were revived after the switch from Wildfly to Quarkus, and the switch from the old to new admin UI. It was both helpful to see historical context, and explain the new functionality to users that had no ability to differentiate “old” from “new” functionality. I can almost write the post for 1-2 years from now about how you’re taking down the Discourse archive because it’s become “too hard” to explain that it represents functionality that no longer exists (e.g. for the “new store”). 

After some reflection and discussion with other community members, I think it comes down to choice of the tool. Discourse has nailed so many aspects of the needs of a community like this, where mailing lists and GitHub Discussions fail. Hard to go into it all, but SEO and read tracking are two huge ones. I worked with a company that had a similarly large community, and after a short bake-off, they quickly concluded that Discourse was the clear winner, primarily because of the search discoverability of content, reputation features and other participant ergonomics.

I’m making the assumption that you (the maintainers and RedHat team) want to facilitate participation of external community members, be they end users, consultants, companies, etc. Having something to show for their participation, such as discoverable, permanent content and reputation tracking, are key incentives. Some products like StackOverflow and Discourse have succeeded because of an understanding of this network effect. If my assumption about your goal is correct, I would suggest selecting a tool that drives as much community participation as possible. It’s support you don’t have to pay for, and builds the appearance and reality of community scale, which causes a virtuous cycle of growth. 

Regarding Slack, I’m assuming you’re obligated to use it by the CNCF. I’ve spent a few days there, and it seems to make the problem of ephemeral content disappearing worse. People come there seeking a quick/live answer, maybe they get a brief answer, but it almost immediately goes down the memory hole once it has scrolled out of view. The company I mentioned above did maintain their Slack, but only because they installed a plug-in that allowed them to “export” threads to Discourse for permanent search discovery, and longer-form answers. Would you be willing to try a similar thing?

I sometimes notice that feature/support requests here are shut down (at least by one maintainer) for the crime of mentioning they work for a company (*gasp*). They are told to “put up” and do it themselves, because they should have the resources to do so. It always makes me laugh, because, honestly, who is configuring an IdP as a hobby? But mostly it makes me realize how little direction most people are given for how to constructively participate. This really gets to the core of my interest, of which choosing a community tool is only a sideshow:

I came to the Keycloak community after several successful implementations at employers. I began consulting for corporate users of Keycloak between jobs. I started building open source extensions, some for customers, others to fill holes I discovered through the Discourse forum. I turned the consulting business into a support and hosting business. We believe that by building an ecosystem of tools around Keycloak, we make the whole community stronger, and grow the pie for everyone. Until now, one of the ways I and our employees have “put up” is to participate in the Discourse forum with quality, respectful content and answers. However, none of the tools that will remain after Monday really give us the same incentives to continue. Given that, **what is your recommendation for engagement?**

**Comment by weltonrodrigo** [comment]
I have been feeling confusion about the Keycloak community. We have too many options and closing one of them seems like a reasonable action.

I also have benefitted from Keycloak immensely and tried to give back answering questions on the forum. Hope I can continue elsewhere.

**Reply by jonkoops** [reply]
Thanks for that @weltonrodrigo, we do very much appreciate community members stepping up to help others. And I hope you will continue to be able to do so on GitHub discussions as well.

**Comment by stianst** [comment]
**Nothing is irrevocable** - if there is wide-spread interest in re-opening Discourse let's see if we can find a way.

I fully appreciate Discourse was an active part of our community, the problem here lies in the fact that it did not get any attention from Keycloak maintainers or other full-time developers on the project, and in the end also had no-one actively moderating or maintaining it.

There's also the challenge of having too many communication channels. I think you can appreciate that we don't have capacity to be part of too many channels.

GitHub Discussions is pretty much a no-brainer for us. It's a channel that we are regardless of Discussions actively taking part in. On a daily basis we are looking at PRs, issues, etc. so there's little extra overhead of having Discussions enabled there.

Slack is more of an ability to talk to individuals, and serves a completely different purpose to a mailing list or a forum in my opinion. Personally for instance there is times I wish I had an easy way to contact individuals in a more engaging way than a forum, and for me Slack brings me this tool.

The mailing lists we've had from day one, and although there is less and less traffic on them, they serve a nice way to send out wider notifications. I see this more of a notification tool, than a discussion tool.

In the end I believe before considering re-opening Discourse we should identify a group of people that can help maintain and moderate it. Maybe that could be fully community driven? If anyone is interested in stepping up to maintain Discourse speak up and let's discuss that as an option.

**Reply by dasniko** [reply]
Hi @stianst , thanks for that clarification. 👍 
I already had a chat in the past with @xgp and we both would step in in some way to maintain/moderate the Discourse forum and we would appreciate any discussion around this! How should we proceed?

**Reply by stianst** [reply]
I'm going on PTO for a week, but let's have a chat about this when I return and we'll figure something out

**Comment by xgp** [comment]
Did the Discourse archive get killed @abstractj ?

<img width="980" alt="Screen Shot 2023-10-03 at 11 13 23 PM" src="https://github.com/keycloak/keycloak/assets/244253/23dfb6a7-5bdc-45f7-91c1-16a9bc3e8561">

Since it's no longer online, is it still possible to get a dump of the content that was there?

**Comment by gotomanners** [comment]
It's back up, and also no longer readonly!
![image](https://github.com/keycloak/keycloak/assets/350858/2f3f2351-da5f-4803-a517-cf8c85cfda86)

**Comment by abstractj** [comment]
@dasniko @xgp @gotomanners the domain was taken down due to inactivity for obvious reasons. I requested to restore the domain. Would you have been interested in moderating the Keycloak discourse instance?

**Reply by dasniko** [reply]
If @xgp still wants to join, I'd be happy to moderate the discourse forum! 🙂

**Reply by abstractj** [reply]
Thank you @dasniko, waiting for others to reply. So we can coordinate it accordingly.

**Reply by xgp** [reply]
@abstractj @dasniko I'm happy to moderate, and I'm glad to see the Discourse coming back!

However, I'm out this week, so apologies if it takes me a long time to reply.

**Reply by abstractj** [reply]
@xgp @dasniko considering that many people from our team are on PTO this week. Let's discuss it next week, to ensure better coordination. Please do not hesitate to reach me out on Slack if you have any questions.

**Comment by abstractj** [comment]
@xgp @dasniko announcement created https://github.com/keycloak/keycloak-web/pull/422. Once it is merged, I will proceed with the next steps.

**Reply by abstractj** [reply]
Published here https://keycloak.discourse.group/t/reactivating-discourse/22931 and https://www.keycloak.org/2023/10/reactivating-discourse.
